webapp/src/main/java/io/github/microcks/service/SpanStorageService.java

```diff
@@ -185,16 +212,18 @@ public static boolean valuesEqualAttr(Object actual, Object expected) {
     * @return Set of trace IDs
     */
    public Set<String> getAllTraceIds() {
-      return spansByTraceId.keySet();
+      synchronized (spansByTraceId) {
+         return Set.copyOf(spansByTraceId.keySet());
```

Comment:
I don't think entrySet() needs to be synchronized as it is a unitary method with no iteration. Also, why would we do a copy here?